presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java

```diff
@@ -657,7 +658,12 @@ public HiveSessionProperties(HiveClientConfig hiveClientConfig, OrcFileWriterCon
                         USE_RECORD_PAGE_SOURCE_FOR_CUSTOM_SPLIT,
                         "Use record page source for custom split",
                         hiveClientConfig.isUseRecordPageSourceForCustomSplit(),
-                        false));
+                        false),
+                integerProperty(
```

Comment:
Consider validating the session property such that `max_initial_splits` must be `>= 0`. I also see that there's currently no check in the `HiveSplitSource` constructor that verifies that value is `>= 0`, and it's probably worth adding that check too while we're at it.